src/Slider/_tests_/index.test.js

```diff
@@ -0,0 +1,488 @@
+    expect(instance.state.value).toEqual([21, 60]);
+    expect(onChange).toHaveBeenCalled();
+  });
+  it("handleTouchMove should not set value when touches > 1", () => {
+    instance.setState({ value: [50, 60] });
+    instance.handleOnTouchMove({
+      touches: [{ pageX: 109 }, { pageX: 110 }],
+    });
+    expect(instance.state.value).toEqual([50, 60]);
+  });
+  it("handleMouseDown should execute onChangeBefore and add listeners", () => {
+    instance.setState({ value: [50, 60] });
+    instance.handleMouseDown(0)({ button: 0, buttons: 1 });
+    expect(typeof eventMap.mousemove === "function").toBe(true);
+    expect(typeof eventMap.mouseup === "function").toBe(true);
+    expect(instance.state.focused).toBe(true);
+    expect(onChangeBefore).toHaveBeenCalled();
+  });
+  it("handleMouseUp should call onChangeAfter and remove listeners", () => {
+    instance.handleMouseUp();
+    expect(typeof eventMap.mousemove !== "function").toBe(true);
+    expect(typeof eventMap.mouseup !== "function").toBe(true);
+    expect(instance.state.focused).toBe(false);
+    expect(onChangeAfter).toHaveBeenCalled();
+  });
+  it("handleMouseMove should call onChange and set value", () => {
+    instance.setState({ value: [50, 60], handleIndex: 0 });
+    instance.handleMouseMove({ pageX: 152 });
+    expect(instance.state.value).toEqual([32, 60]);
+    expect(onChange).toHaveBeenCalled();
+  });
+  test("replaceValue should return number or array number", () => {
+    instance.setState({ value: [50, 60] });
+    expect(instance.replaceValue(10, 1)).toEqual([50, 10]);
+  });
+});
+
+describe("Range Slider in render", () => {
+  const min = 1;
+  const max = 24;
+  const dataTest = "test";
+  const component = shallow(
+    <Slider
+      min={min}
+      max={max}
+      defaultValue={[1, 24]}
+      label="Depart from Prague"
+      dataTest={dataTest}
+    />,
+  );
+  const slider = component.find("Slider__StyledSlider");
+  test("should have data-test", () => {
+    expect(slider.render().prop("data-test")).toBe(dataTest);
+  });
+  test("calculateValueFromPosition should return null", () => {
+    expect(component.instance().calculateValueFromPosition(52)).toBe(null);
+  });
+  test("should have two Handles", () => {
+    const children = slider.find("Slider__StyledSliderInput").find("Handle");
+    children.forEach((node, index) => {
+      expect(node.prop("valueMax")).toBe(max);
+      expect(node.prop("valueMin")).toBe(min);
+      expect(node.prop("value")).toEqual([1, 24]);
+      expect(node.prop("index")).toBe(index);
+    });
+  });
+  test("should have Bar component", () => {
+    const children = slider.find("Slider__StyledSliderInput").find("Bar");
+    children.forEach(node => {
+      expect(node.prop("max")).toBe(max);
+      expect(node.prop("min")).toBe(min);
+      expect(node.prop("value")).toEqual([1, 24]);
+    });
+  });
+  it("should match snapshot", () => {
+    expect(component).toMatchSnapshot();
```

Comment:
snapshot tests should be used sparingly, and I don't see a reason to use it here.
it's just something that will break each time you modify the Slider, but it won't give you any real information.

Reply:
For now, the standard is to have it for each component, so we should keep it that way. We will decide what to do with snapshots lates 🙂